common/vcomputer-system.cpp

```diff
@@ -285,4 +308,39 @@ namespace tec {
 			}
 		}
 	}
+	void VComputerSystem::On(std::shared_ptr<EntityCreated> data) {
+		eid entity_id = data->entity.id();
+		for (int i = 0; i < data->entity.components_size(); ++i) {
+			const proto::Component& comp = data->entity.components(i);
+			switch (comp.component_case()) {
+				case proto::Component::kComputer:
+				{
+					Computer* computer = new Computer();
+					computer->In(comp);
+					ComputerComponentMap::Set(entity_id, computer);
+				}
+				break;
+				case proto::Component::kCollisionBody:
+				case proto::Component::kRenderable:
+				case proto::Component::kPosition:
+				case proto::Component::kOrientation:
+				case proto::Component::kView:
+				case proto::Component::kAnimation:
+				case proto::Component::kScale:
+				case proto::Component::kVelocity:
+				case proto::Component::kAudioSource:
+				case proto::Component::kPointLight:
+				case proto::Component::kDirectionalLight:
+				case proto::Component::kSpotLight:
+				case proto::Component::kVoxelVolume:
+				case proto::Component::kLuaScript:
+				case proto::Component::COMPONENT_NOT_SET:
+				break;
```

Comment:
Same here: can we register for kComputer-added specifically?

Reply:
The way that protobuf defines components require us to iterate over all of them, but we could add a switch in another location that sends out specific events.

This would make a great issue as it seems a bit out of scope for this PR.